src/parser/newast.c

```diff
@@ -72,6 +93,39 @@ static const cypher_astnode_t *_NEWAST_GetClause(const cypher_astnode_t *node, c
     return NULL;
 }
 
+/* Collect references to all MATCH clauses in the query. Since clauses cannot be nested, we only need to
```

Comment:
Hm, operations like `MERGE...SET` have no nesting, but `MERGE...ON MATCH` does.

`ON MATCH`/`ON CREATE` are defined as "actions" rather than clauses, but practically speaking they seem identical to a `SET` clause.